libsolidity/parsing/Parser.cpp

```diff
@@ -242,14 +245,21 @@ ASTPointer<ImportDirective> Parser::parseImportDirective()
 	return nodeFactory.createNode<ImportDirective>(path, unitAlias, move(symbolAliases));
 }
 
-ContractDefinition::ContractKind Parser::parseContractKind()
+ContractDefinition::ContractKind Parser::parseContractKind(bool *_isAbstract)
 {
 	ContractDefinition::ContractKind kind;
+	*_isAbstract = false;
 	switch(m_scanner->currentToken())
 	{
 	case Token::Interface:
 		kind = ContractDefinition::ContractKind::Interface;
 		break;
+	case Token::Abstract:
+		*_isAbstract = true;
+		kind = ContractDefinition::ContractKind::Contract;
+		if (Token::Contract != m_scanner->next())
+			fatalParserError(string("Only contracts can be abstract."));
```

Comment:
This should not be a fatal error. Just assume it is not abstract in the other case.